framework/utils/fmuExporter.py

```diff
@@ -0,0 +1,193 @@
+FilePath = Union[str, Path]
+
+class FMUexporter(MessageUser):
+  """
+    FMU exporter for RAVEN
+  """
+  def __init__(self, **kwargs):
+    """
+      FMU builder constructor
+      @ In, **kwargs, dict, kwarded dictionary with the builder options
+      @ Out, None
+    """
+    #instanciate message user
+    super().__init__()
+    # grep options
+    self._options = kwargs
+    # keep the module
+    self.keepModule = self._options.pop("keepModule", False)
+    # check working dir
+    self.workingDir = self._options.pop("workingDir", None)
+    if self.workingDir is None and self.keepModule:
+      self.raiseAnError(IOError, "No workingDir has been provided for FMU exporter!")
+
+    self.model = self._options.pop("model", None)
+    if self.model is None:
+      self.raiseAnError(IOError, "No model has been provided for FMU exporter!")
+    # temp folder
+    if self.keepModule:
+      self._temp = self.workingDir
+    else:
+      self._temp = tempfile.TemporaryDirectory(prefix="raven_fmu_")
+    # we serialize the model and this model will be stored in the resource folder
+    self.serializedModel = Path(self._temp) / (self.model.name + ".pk")
+    # picklefile in temp directory
+    with open(self.serializedModel, mode="wb+") as pk:
+      cloudpickle.dump(self.model, pk, protocol=pickle.HIGHEST_PROTOCOL)
+    self.executeMethod = self._options.pop("executeMethod", None)
+    if self.executeMethod is None:
+      self.raiseAnError(IOError, "No executeMethod has been provided for FMU exporter!")
+    self.inputVars = self.model._getVariableList('input')
+    self.outVars =  self.model._getVariableList('output')
+    if not self.inputVars or not self.outVars:
+      self.raiseAnError(IOError, "Model {} is not exportable as FMU since no info about inputs/outputs are available!".format(self.model.name))
+    self.indexReturnDict = self._options.pop("indexReturnDict", None)
+    if self.indexReturnDict is None:
+      self.raiseAMessage("No indexReturnDict has been provided for FMU exporter! Default to 0!")
+      self.indexReturnDict = 0
+    self.frameworkDir = self._options.pop("frameworkDir")
+    if self.frameworkDir is None:
+      self.raiseAnError(IOError, "No frameworkDir has been provided for FMU exporter!")
+
+  def createModelHandler(self):
+    """
+      This create a temporary file that represents the model to export
+      @ In, None
+      @ Out, createModelHandler, str, the module for the FMU creation
+    """
+    classname = self.model.name.upper()
+    filename = self.model.name + ".pk"
+    # inputVars = ['{}'.format(el) for el in self.inputVars]
+    # outVars = ['{}'.format(el) for el in self.outVars]
+
+    return f"""
+import sys
+import pickle
+import os
+from pythonfmu.fmi2slave import Fmi2Type, Fmi2Slave, Fmi2Causality, Fmi2Variability, Integer, Real, Boolean, String
+
+class {classname}(Fmi2Slave):
+  #
+  #  RAVEN (raven.inl.gov) Model-based Python-driven simulator
+  #
+  author = "Andrea Alfonsi"
```

Comment:
@aalfonsi Do you want to keep your name here? I think this file is mainly for raven users.

Reply:
Oh I would just modify it to author = "RAVEN team"

Reply:
Changed.